src/java/org/apache/cassandra/service/accord/serializers/IVersionedWithKeysSerializer.java

```diff
@@ -171,6 +171,36 @@ else if (supersetCount < 64)
             }
         }
 
+        @DontInline
+        private <K extends Routable, R extends Routables<K>> long serializeLargeSubsetSize(R serialize, int serializeCount, R superset, int supersetCount)
+        {
+            long size = TypeSizes.sizeofUnsignedVInt(supersetCount - serializeCount);
+            if (serializeCount == 0) return size;
+            int supersetIndex = 0;
+            int take = 0;
+            for (int i = 0; i < serializeCount; i++)
+            {
+                int offset = supersetIndex + take;
+                int nextIndex = superset.findNext(offset, serialize.get(i), FAST);
+                if (nextIndex == offset)
+                {
+                    take++;
+                    continue;
+                }
+                if (take != 0) // since this is dealing with subsets, the only time take=0 is when i=0 and the first superset offset isn't included
+                {
+                    size += TypeSizes.sizeofUnsignedVInt(take);
+                    size += TypeSizes.sizeofUnsignedVInt(supersetIndex);
+                }
```

Comment:
this was also there in the original, which leads to a dummy iteration that takes 2 bytes and doesn't do anything.